src/soc/ibm/power9/istep_13_2.c

```diff
@@ -0,0 +1,180 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include <cpu/power/istep_13.h>
+#include <console/console.h>
+#include <cpu/power/scom.h>
+#include <cpu/power/scom_registers.h>
+#include <timer.h>
+
+#define MCS_PER_PROC		2
+#define DIMMS_PER_MCS		4
+
+/*
+ * 13.2 mem_pll_reset: Reset PLL for MCAs in async
+ *
+ * a) p9_mem_pll_reset.C (proc chip)
+ *    - This step is a no-op on cumulus as the centaur is already has its PLLs
+ *      setup in step 11
+ *    - This step is a no-op if memory is running in synchronous mode since the
+ *      MCAs are using the nest PLL, HWP detect and exits
+ *    - If in async mode then this HWP will put the PLL into bypass, reset mode
+ *    - Disable listen_to_sync for MEM chiplet, whenever MEM is not in sync to
+ *      NEST
+ */
+void istep_13_2(dimm_attr dimms[CONFIG_DIMM_MAX])
+{
+	printk(BIOS_EMERG, "starting istep 13.2\n");
+	int i;
+	long time_elapsed = 0;
+	bool mcs_functional[MCS_PER_PROC] = {false, false};
+	chiplet_id_t mcs_ids[MCS_PER_PROC] = {MC01_CHIPLET_ID, MC23_CHIPLET_ID};
+
+	report_istep(13,2);
+
+	for (i = 0; i < CONFIG_DIMM_MAX; i++) {
+		if (dimms[i].dram_type == SPD_MEMORY_TYPE_DDR4_SDRAM &&
+		    dimms[i].dimm_type == SPD_DIMM_TYPE_RDIMM &&
+		    dimms[i].ecc_extension == true) {
+			mcs_functional[i/DIMMS_PER_MCS] = true;
+			/* Skip to next MCS */
+			i = (i/DIMMS_PER_MCS + 1) * DIMMS_PER_MCS;
+		}
+	}
+
+	/*
+	 * There is only one MCBIST per CPU. Fail if there are no supported DIMMs
+	 * connected, otherwise assume it is functional. There is no reason to redo
+	 * this test in the rest of isteps.
+	 *
+	 * TODO: 2 CPUs with one DIMM (in total) will not work with this code.
+	 */
+	if (mcs_functional[0] == false && mcs_functional[1] == false)
+		die("No DIMMs detected, aborting\n");
+
+	/* Assuming MC doesn't run in sync mode with Fabric, otherwise this is no-op */
+
+	for (i = 0; i < MCS_PER_PROC; i++) {
+		if (!mcs_functional[i])
+			continue;
+
+		// Assert endpoint reset
+		/*
+		TP.TPCHIP.NET.PCBSLMC01.NET_CTRL0 (WOR)                 // 0x070F0042
+		  [all] 0
+		  [1]   PCB_EP_RESET =  1
+		*/
+		write_scom_for_chiplet(mcs_ids[i], 0x070F0042, PPC_BIT(1));
```

Comment:
Let's make defines for registers instead of pasting the hex values. It should increase the readability, and reduce the number of lines of the comments. The same goes for PCB_EP_RESET and other bits used here